src/lucky/http2_app_server.cr

```diff
@@ -0,0 +1,94 @@
+# HTTP2 App Server for Lucky applications
+#
+# This class provides HTTP/2 support using the ht2 shard.
+# It requires TLS certificates and is enabled via the http2_enabled setting.
+{% if flag?(:http2) %}
```

Comment:
I'm not sure if we have access to pass this flag 🤔  When you run `lucky dev`, that tells the CLI to call Nox which runs the `Procfile.dev` that calls `lucky watch` that internally calls `crystal build`

https://github.com/luckyframework/lucky/blob/ff904b28c7a0fdcd45148be3850cc00d5984bb59/tasks/watch.cr#L331

I'd have to test, but I'm not sure if doing `lucky dev -Dhttp2` would ever get picked up by that build.